### ISSUE TITLE
Good code, but a little problem

### ISSUE THREAD

**Issue by JinFish** [chenjunyi1999/Translator-Transformer#1]
I read your code, and I think it's very good.
However, I found that the code in line 215 and line 220 is same: scores = torch.matmul(query, key.transpose(-2, -1) / math.sqrt(d_k))

**Comment by chenjunyi1999** [comment]
haha ^_^.  thank you for pointing it out.

**Comment by JinFish** [comment]
Thank you for your response.
Another question:  In the computing of loss, `true_dist.index_fill_(0, mask.squeeze(), 0.0)` means that we do not calculate the loss of `[PAD]` (I think). Should we also ignore the loss of `[EOS]`?
```python
class LabelSmoothing(nn.Module):
    def __init__(self, size, padding_idx, smoothing=0.0):
        super(LabelSmoothing, self).__init__()
        # 损失函数 nn.KLDivLoss / KL散度，又叫做相对熵，算的是两个分布之间的距离，越相似则越接近零。
        self.criterion = nn.KLDivLoss(reduction='sum')
        self.padding_idx = padding_idx
        self.confidence = 1.0 - smoothing
        self.smoothing = smoothing
        self.size = size
        self.true_dist = None

    def forward(self, x, target):
        assert x.size(1) == self.size
        # 如果我们想要修改tensor的数值，但是又不希望被autograd记录，那么我么可以对tensor.data进行操作
        true_dist = x.data.clone()
        # [个人理解！] 减去UNK 和target的idx 之后取均匀分布 smoothing/(n-2)
        true_dist.fill_(self.smoothing / (self.size - 2))
        # 将target的idx在one-hot中位置改成大小为confidence 0.9 (1-smoothing)
        true_dist.scatter_(1, target.data.unsqueeze(1), self.confidence)
        # 将UNK这一列置0
        true_dist[:, self.padding_idx] = 0
        mask = torch.nonzero(target.data == self.padding_idx)
        if mask.dim() > 0:
            true_dist.index_fill_(0, mask.squeeze(), 0.0)
        self.true_dist = true_dist
        return self.criterion(x, Variable(true_dist, requires_grad=False))
```

**Comment by chenjunyi1999** [comment]
Really sorry for the late reply.
I believe that [EOS] should not be ignored, as it serves as a sentence ending marker, helping the model learn when to terminate generation.
If my reply is incorrect, please feel free to point it out. (actually this is my code from four years ago. and I haven't been working in this field for a long time)  
Hope everything works out for you.

**Comment by JinFish** [comment]
I think you are right. And I'm sorry to bother you.
